### PR TITLE
Connect with bs-fullscreen-message

### LINKED ISSUE #25: [Question] How to setup bs-fullscreen-message
I'm trying to setup `bs-fullscreen-message` browser sync plugin with this webpack plugins, but it looks that doesn't work.

Any tip about how to setup correctly?

Related: https://github.com/shakyShane/bs-fullscreen-message#usage

### PR DESCRIPTION
**Pull request by Kikobeats**
Fix #25 

Full screen errors support

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   ],
   "main": "index.js",
   "dependencies": {
-    "bs-fullscreen-message": "~1.1.0",
+    "bs-fullscreen-message": "Kikobeats/bs-fullscreen-message",
```
Comment (Kikobeats) [inline package.json]: I linked my fork because original code looks unmaintained https://github.com/shakyShane/bs-fullscreen-message/pull/5

File: index.js [hunk 2]
```diff
@@ -25,10 +25,10 @@ BrowserSyncPlugin.prototype.apply = function (compiler) {
...
     if (stats.hasErrors() || stats.hasWarnings()) {
+      var error = stats.toString('minimal')
       return self.browserSync.sockets.emit('fullscreen:message', {
         title: 'Webpack Error:',
-        body: stripAnsi(stats.toString()),
-        timeout: 100000
+        body: stripAnsi(error),
       })
     }
     self.browserSync.reload()
```
Comment (tchiotludo) [inline index.js]: IMO, it's better to like this : 
```js
if (stats.hasErrors() || stats.hasWarnings()) {
  // ...
} else {
  self.browserSync.reload()
}
```
With this options, if we add an option `{reload: false)`, it prevent reloading page on error.
Reload is too fastest to see the proper error message.

I also like to have an option for timeout, default timeout is too short in my opinion.

Comment (Kikobeats) [inline index.js]: Let me try your approach, thanks!

Actually I removed the timeout part because I don't found the sense of show an error with timeout; I'm going to update this.

Comment (tchiotludo) [inline index.js]: I agree with you for timeout part, IMO, it can be hard coded to infinite timeout.
Personally, if you want to remove the error message, I fix my error :smile:

Comment (Kikobeats) [inline index.js]: the point is, when  the error is fixed, the fullscreen message disappear, so not necessary timeout!

**Comment by Kikobeats** [comment]
I reimplemented `bs-fullscreen-message` for avoid use timeout. Then I connect the library with the compile events for clean or show the error.

Things about `browser-sync-webpack-plugin` code here:

- I added `strip-ansi` dep for clean ansi codes. Teorically could be possible disable colors passing an object to `.toString` with `colors:false` but I don't know doesn't work. You can investigate [here](https://github.com/webpack/webpack/blob/c08d1b4f89aa16b65bbcf4b583efbe0c83322e77/lib/Stats.js#L738) and [here](https://webpack.github.io/docs/node.js-api.html#stats-tostring).

- I recommend you, instead of load lodash for use two methods, install only the lodash dependencies that you need, like [lodash.isfunction](https://www.npmjs.com/package/lodash.isfunction) and [lodash.asiggin](https://www.npmjs.com/package/lodash.assignin) (the lodash name for _.extend).

- Notes how the plugin is connected with the webpack errors. Maybe exists more webpack errors that we can consider, but actually I don't know. I noticed that if you run webpack first time and build was failed, message will not displayed; need a start success build for start work.

**Comment by Kikobeats** [comment]
oh, one thing more: with my code, `compiler.plugin('done'` is declared twice times; maybe could be possible merge in one function.

**Comment by Kikobeats** [comment]
@Va1 are you alive?

**Comment by Va1** [comment]
@Kikobeats well, not really :D
though, to be serious, i promise to look into the issue and pr today/tomorrow. excuse me for delaying this for so long time

**Comment by Va1** [comment]
@Kikobeats 
OK well, I had a look into the related issue, proposed changes and into the package itself and here's what I can say about this:

1) I do not feel like the functionality of this package should be extended beyond what it is now, it is pretty complete. another package, which provides fancy messages in browser, is a cool thing for the ones who want it, but it is **another** feature and it makes this package do **two** things instead of **one** if I merge your PR (you know, unix way, blah-blah, everything should do only one thing and do it wll, blah-blah-blah). what do you think?

2) after looking into this package code, then into that package usage (you gave link for), I do not see any reasons for this integration to work without merging two packages. This plugin accepts the browser-sync options and passes em directly to the ```browserSync.init(...)```, so you can add ```plugins: ['bs-fullscreen-message']``` or whatever needed from configuration side. why this approach did not work for you?

3) regarding lodash - fair enough, but, as you know, lodash is the most-required package on npm and pretty every (absolute majority, I think) of frontend build configurations (I assume that this package is being used only for building and working with frontend stuff) require lodash, so npm will not load yet another copy of lodash into package node_modules because of flattening and optimization mechanisms. Though, this is  quite fair as I said, and I will be glad if you submit a separate PR for this.

To sum up, I do not think that having this code merged is a good idea.

**Comment by Kikobeats** [comment]
np, I can create a different package
